smolyay/surrogate.py

```diff
@@ -631,6 +635,37 @@ def fit_gradient(self, X, y, X0=None, y0=None):
             self._integration_constant = integration_constant
         return self
 
+    def compute_definite_integral(self):
+        """Computes the definite integral of the surrogate
+
+        Returns
+        -------
+        float
+            definite integral
+        """
+        lookup_table = [
+            self.basis_sets[dim].integral_over_domain
+            for dim in range(self.num_dimensions)
+        ]
+        # definite integral of integration constant
+        answer = numpy.sum(
+            [
+                self._integration_constant * (bs.domain[1] - bs.domain[0])
+                for bs in self.basis_sets
+            ]
+        )
+        # use lookup table to combine terms
+        for ic, coeff in zip(self._index_combinations, self._coefficients):
+            answer += numpy.real(
+                coeff
+                * numpy.prod(
+                    [lookup_table[dim][ic[dim]] for dim in range(len(ic))], axis=0
+                )
+            )
+
+        # return results
+        return answer
```

Comment:
Maybe I am missing it, but I don't think I'm seeing a spot where the integral gets multiplied by a factor accounting for the domain of the surrogate being different than that of the basis functions? You should be able to take your answer as you've computed it and multiply it by this factor before you return it.

I also don't think I see a unit test covering this case.

Reply:
I hadn't added that feature yet. It's added now.